### PR TITLE
Sync the relations with the model relation list

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3548.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add a bundle with a few apps/relations
- click Destroy on one of the apps
- click on the inspector for one of the apps that used to relate to the app you destroyed
- open the relation list
- you should not get errors in your console
- click export in the header
- the exported relation list should not include any of the relations to the destroyed app

**Comment by huwshimi** [comment]
[Fix removed relations when exporting models](https://trello.com/c/dAd365F1/368-fix-removed-relations-when-exporting-models)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8551/

**Comment by huwshimi** [comment]
@frankban thanks, tests added.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8552/

File: jujugui/static/gui/src/app/init.js [hunk 1]
```diff
@@ -318,6 +321,25 @@ class GUIApp {
     }
   }
 
+  /**
+    Sync relation changes to the app relation stores.
```
Comment (hatched) [inline jujugui/static/gui/src/app/init.js]: can you add a 'why' to this comment? I don't fully understand it and I know the issue :D

**Comment by huwshimi** [comment]
I'm going to close this because you're probably right, but that's a bigger issue that needs to be resolved in the future.

**Comment by huwshimi** [comment]
Oops, closed the wrong branch.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8559/

**Comment by huwshimi** [comment]
Superseded by #3574.

**Pull request closed without merging**